test/Unit/Task/ESLintTest.php

```diff
@@ -0,0 +1,121 @@
+
+namespace GrumPHP\Test\Unit\Task;
+
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use GrumPHP\Task\ESLint;
+use GrumPHP\Task\TaskInterface;
+use GrumPHP\Test\Task\AbstractExternalTaskTestCase;
+
+class ESLintTest extends AbstractExternalTaskTestCase
+{
+    protected function provideTask(): TaskInterface
+    {
+        return new ESLint(
+            $this->processBuilder->reveal(),
+            $this->formatter->reveal()
+        );
+    }
+
+    public function provideConfigurableOptions(): iterable
+    {
+        yield 'defaults' => [
+            [],
+            [
+                // Task config options
+                'bin' => null,
+                'triggered_by' => ['js', 'jsx', 'ts', 'tsx', 'vue'],
+                'whitelist_patterns' => null,
+
+                // ESLint native config options
+                'config' => null,
+                'debug' => false,
+                'format' => null,
+                'max_warnings' => null,
+                'no_eslintrc' => false,
+            ]
+        ];
+    }
+
+    public function provideRunContexts(): iterable
+    {
+        yield 'run-context' => [
+            true,
+            $this->mockContext(RunContext::class)
+        ];
+
+        yield 'pre-commit-context' => [
+            true,
+            $this->mockContext(GitPreCommitContext::class)
+        ];
+
+        yield 'other' => [
+            false,
+            $this->mockContext()
+        ];
+    }
+
+    public function provideFailsOnStuff(): iterable
+    {
+        yield 'exitCode1' => [
+            [],
+            $this->mockContext(RunContext::class, ['hello.js']),
+            function () {
+                $this->mockProcessBuilder('eslint', $process = $this->mockProcess(1));
+                $this->formatter->format($process)->willReturn('nope');
+            },
+            'nope'
+        ];
+    }
+
+    public function providePassesOnStuff(): iterable
+    {
+        yield 'exitCode0' => [
+            [],
+            $this->mockContext(RunContext::class, ['hello.js']),
+            function () {
+                $this->mockProcessBuilder('eslint', $this->mockProcess(0));
+            }
+        ];
+    }
+
+    public function provideSkipsOnStuff(): iterable
+    {
+        yield 'no-files' => [
+            [],
+            $this->mockContext(RunContext::class),
+            function () {}
+        ];
+        yield 'no-files-after-triggered-by' => [
+            [],
+            $this->mockContext(RunContext::class, ['notajsfile.txt']),
+            function () {}
+        ];
+        yield 'no-files-after-whitelist-patterns' => [
+            [
+                'whitelist_patterns' => ['/^resources\/js\/(.*)/'],
+            ],
+            $this->mockContext(RunContext::class, ['resources/dont/find/this/file.js']),
+            function () {}
+        ];
+    }
+
+    public function provideExternalTaskRuns(): iterable
+    {
+        yield 'config' => [
```

Comment:
The goal from the provider is to test every possible CLI command based on the task options.
You basically need to add a state for every possible option:

```
        $arguments = isset($config['bin'])
            ? ProcessArgumentsCollection::forExecutable($config['bin'])
            : $this->processBuilder->createArgumentsForCommand('eslint');

        $arguments->addOptionalArgument('--config=%s', $config['config']);
        $arguments->addOptionalArgument('--debug', $config['debug']);
        $arguments->addOptionalArgument('--format=%s', $config['format']);
        $arguments->addOptionalArgument('--no-eslintrc', $config['no_eslintrc']);
        $arguments->addOptionalIntegerArgument('--max-warnings=%d', $config['max_warnings']);
        $arguments->addFiles($files);
```

In this case, you covered the line 

```
        $arguments->addOptionalArgument('--config=%s', $config['config']);
```

Reply:
Done, now seeing 100% code coverage on the ESLint class ✅